src/LocalFeedConfigs.php

```diff
@@ -0,0 +1,119 @@
+<?php //phpcs:disable WordPress.WP.AlternativeFunctions --- Uses FS read/write in order to reliable append to an existing file.
+/**
+ * Class responsible for managing local feed configurations.
+ *
+ * @package     Pinterest_For_WooCommerce/Classes/
+ * @version     x.x.x
+ */
+namespace Automattic\WooCommerce\Pinterest;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Class Handling feed files generation.
+ *
+ * Singleton pattern used.
+ * Prevent application from having multiple instances of configuration.
+ * At the same time allow distributed use of configurations.
+ */
+class LocalFeedConfigs {
+
+	/**
+	 * Array of local feed configurations.
+	 *
+	 * @var array $feeds_configurations
+	 */
+	private $feeds_configurations = null;
+
+	/**
+	 * The Singleton's instance.
+	 *
+	 * @since x.x.x
+	 * @var null|LocalFeedConfigs Instance object.
+	 */
+	private static $instance = null;
+
+	/**
+	 * Singleton initialization and instance fetching method.
+	 *
+	 * @since x.x.x
+	 * @return LocalFeedConfigs Singleton instance.
+	 */
+	public static function get_instance(): LocalFeedConfigs {
+		$cls = static::class;
+		if ( null === self::$instance ) {
+			self::$instance = new static();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Class responsible for local feed configurations and handling.
+	 *
+	 * @since x.x.x
+	 */
+	protected function __construct() {
+		$locations = array( Pinterest_For_Woocommerce()::get_base_country() ?? 'US' ); // Replace with multiple countries array for multiple feed config.
+		$this->initialize_local_feeds_config( $locations );
+	}
+
+	/**
+	 * Prepare feed configurations.
+	 *
+	 * @since x.x.x
+	 * @param array $locations Array of location to generate the feed files for.
+	 */
+	private function initialize_local_feeds_config( $locations ) {
+
+		if ( $this->feeds_configurations ) {
+			return;
+		}
```

Comment:
This method is only called from the constructor, and is `private`. I don't think we need this check here.

```suggestion
```

Reply:
That is a nice simplification. The singleton patter was added in one of the last commits and made this part redundant. Cool, removing.